### PR TITLE
Sample RBAC for more fine grained security

### PR DESCRIPTION
**Pull request by SudharmaPuranikLeanIX**
## This PR is to just to showcase the RBAC flex! Not intended for PROD

1. ClusterRole with only details on what we need.
2. `binding.yaml` is the RoleBinding which Cariad would take care.

### CODE REVIEW

**Review by QuentinVaucherLeanIX — COMMENTED** [review]
You might have missed it as you were not yet back in Helios but: all your commits need to be prefixed by a jira ticket number. You can read @DennisBerthold-leanix post here: https://leanix.slack.com/archives/C02P1EZT6TX/p1687342267624499

**Comment by SudharmaPuranikLeanIX** [comment]
> You might have missed it as you were not yet back in Helios but: all your commits need to be prefixed by a jira ticket number. You can read @DennisBerthold-leanix post here: https://leanix.slack.com/archives/C02P1EZT6TX/p1687342267624499

Thanks @QuentinVaucherLeanIX thats a good point. :) I missed adding my pre-commit hook

File: helm/leanix-k8s-connector/templates/clusterrole.yaml [hunk 1]
```diff
@@ -7,42 +7,12 @@ metadata:
...
-- apiGroups: ["storage.k8s.io"]
-  resources:
-  - storageclasses
-  verbs:
-  - get
-  - list
-  - watch
-{{- end -}}
\ No newline at end of file
+  resources: ["nodes","persistentvolume","persistentvolumeclaim","namespaces","services"]
```
Comment (daniellesarmiento) [inline helm/leanix-k8s-connector/templates/clusterrole.yaml:10]: For my understanding...do adding these resources in the list make permissions more fine-grained? Doesn't the node have an "overview" of all these resources?

Comment (SudharmaPuranikLeanIX) [reply helm/leanix-k8s-connector/templates/clusterrole.yaml]: @daniellesarmiento From the API perspective https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#node-v1-core for `[get,list,watch]` it does not consider these resources. However even if we use `delete` the API server takes care of it. Thats the abstraction of  REST API operation w.r.t the actions on the resources.

**Comment by QuentinVaucherLeanIX** [comment]
@SudharmaPuranikLeanIX I see more commits without the `HEL-` prefix. Could you rewrite your commit history by adding the prefix, please? 🙏🏼

**Comment by QuentinVaucherLeanIX** [comment]
> > You might have missed it as you were not yet back in Helios but: all your commits need to be prefixed by a jira ticket number. You can read @DennisBerthold-leanix post here: https://leanix.slack.com/archives/C02P1EZT6TX/p1687342267624499
> 
> Thanks @QuentinVaucherLeanIX thats a good observation. :) BTW this is just a spike and not attached with ticket. I have already pre-commit hook on to take care of the prefix.
Afaik, even a spike should have a ticket number, since any code changes need to be link to a jira ticket. Maybe @DennisBerthold-leanix can confirm/invalidate this

**Pull request closed without merging**